### PR TITLE
mon: remove pre-luminous health output compatibility

### PR DESCRIPTION
**Pull request by jcsp**
As of nautilus, this will be more than two versions old:
external tooling should have been updated by now.

Signed-off-by: John Spray <john.spray@redhat.com>

### CODE REVIEW

**Comment by jcsp** [comment]
My motivation here is partly code cleanup, and partly to make the JSON health output a bit friendlier to human eyes by removing the old fields and the big warning message.

**Comment by tchaikov** [comment]
@jcsp i think this change breaks http://pulpito.ceph.com/kchai-2018-08-24_03:54:41-rados-wip-kefu2-testing-2018-08-23-2301-distro-basic-smithi/2940450/ .

```
2018-08-24T06:02:15.190 INFO:tasks.workunit.client.0.smithi122.stdout:{"checks":{"OSD_OUT_OF_ORDER_FULL":{"severity":"HEALTH_ERR","summary":{"message":"full r
atio(s) out of order"}}},"status":"HEALTH_ERR"}
2018-08-24T06:02:15.191 INFO:tasks.workunit.client.0.smithi122.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:1883: test_mon_pg:  c
eph health detail
2018-08-24T06:02:15.191 INFO:tasks.workunit.client.0.smithi122.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:1883: test_mon_pg:  g
rep OSD_OUT_OF_ORDER_FULL
2018-08-24T06:02:15.560 INFO:tasks.workunit.client.0.smithi122.stderr:/home/ubuntu/cephtest/clone.client.0/qa/workunits/cephtool/test.sh:1: test_mon_pg:  rm -
fr /tmp/cephtool.Yg3
```

seems `OSD_OUT_OF_ORDER_FULL` failed to show up in the output `ceph health detail` now.

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -2579,43 +2578,6 @@ health_status_t Monitor::get_health_status(
...
-    if (f && (compat || compat_warn)) {
-      f->open_array_section("detail");
-      if (compat_warn) {
-	f->dump_string("item", "'ceph health' JSON format has changed in luminous. If you see this your monitoring system is scraping the wrong fields. Disable this with 'mon health preluminous compat warning = false'");
-      }
-    }
-
-    for (auto& svc : paxos_service) {
-      svc->get_health_checks().dump_detail(f, plain, compat);
-    }
```
Comment (liewegas) [inline src/mon/Monitor.cc]: This block shouldn't get removed

Comment (jcsp) [reply src/mon/Monitor.cc]: ah, I missed the plain text handling in that fn.  Reinstated now.

**Pull request merged**